fabtests/functional/rdm_stress.c

```diff
@@ -593,58 +592,280 @@ static int run_child(void)
 
 	for (i = 0; i < ctrl_cnt && !ret; i++) {
 		printf("(%d-%d) rpc op %s\n", myid, id_at_server,
-		       rpc_op_str(ctrl[i].op));
-		ret = ctrl_op[ctrl[i].op](&ctrl[i]);
+		       rpc_op_str(ctrls[i].op));
+		ret = ctrl_op[ctrls[i].op](&ctrls[i]);
 	}
 
 free:
 	ft_free_res();
 	return ret;
 }
 
+static bool get_uint64_val(const char *json_payload, jsmntok_t *token,
+			   uint64_t *val)
+{
+	if (token->type != JSMN_PRIMITIVE)
+		return false;
 
-static struct rpc_ctrl ctrl_array[] = {
-	{op_goodbye},
-	{op_hello},
-	{op_msg_req, 1000},
-	{op_msg_resp},
-	{op_tag_req, 2000},
-	{op_tag_resp},
-	{op_msg_req, 1000000},
-	{op_msg_resp},
-	{op_write_req, 5600000, {12000}},
-	{op_write_resp},
-	{op_read_req, 64000},
-	{op_read_resp},
-	{op_tag_req, 2000000},
-	{op_tag_resp},
-	{op_read_req, 32000},
-	{op_read_resp},
-	{op_write_req, 86000, {6000}},
-	{op_write_resp},
-	{op_read_req, 1000000},
-	{op_read_resp},
-	{op_write_req, 56000, {12000}},
-	{op_write_resp},
-	{op_sleep, 100},
-	{op_exit},
-};
+	return (sscanf(&json_payload[token->start], "%lu", val) == 1);
+}
 
-/* TODO: read and parse control file */
-static int init_rpc(void)
+static bool get_op_enum(const char *json_payload, jsmntok_t *token,
+			uint32_t *op)
 {
-	ctrl = ctrl_array;
-	ctrl_cnt = ARRAY_SIZE(ctrl_array);
+	const char *enum_val_str;
+	size_t enum_val_len;
+
+	if (token->type != JSMN_STRING)
+		return false;
+
+	enum_val_str = &json_payload[token->start];
+	enum_val_len = token->end - token->start;
+
+	if (!strncmp(enum_val_str, "msg_req", enum_val_len)) {
```

Comment:
hmm.  There is a problem with length calculation, off by 1.  I got lucky with order of comparison. Will fix that.
json_payload is just a long string with 1 null at the end.  Given the length parameter, we will always compare to the full enum_val_len.  I will review order of comparison to make sure I'm not getting lucky and be less dependent on the length parameter.

Reply:
At least it would be better to use the length of "msg_req" (7) instead of `enum_val_len` here.  In order to get exact match comparison, the two lengths must be equal. Could add a utility function for such comparison.

Reply:
I double checked start and end pointer.  end is actually a character after the string.  For example, with "msg_req" string, start points to 'm' and end points to the 2nd ".  So the math is correct given where the pointers are pointing.

Reply:
I ended up reordering strncmps.  Checking shorter strings first.